### DISCUSSION TITLE
Start two different session with janus client sometimes its give same preview in both the preview layer

### DISCUSSION THREAD

**Discussion by ChiragLiveu** [flutterjanus/flutter_janus_client#116] [Q&A]
I have run two separate clients using two separate channels and web IDs but sometimes in the preview layer, I will get the same preview in both the instance in iOS & Android as well. I have debugged and checked each time I have passed two different webID to stream on the server side it takes two different requests and sends a proper preview but on the mobile side, it takes the same preview layer sometimes. I am using the streaming through a web socket take reference from the below link.

https://github.com/flutterjanus/flutter_janus_client/blob/master/example/lib/typed_examples/streaming.dart

**Comment by shivanshtalwar0** [comment]
is it still happening?

**Reply by ChiragLiveu** [reply]
Yes I  have the still same problem can you please help me for the same.

**Reply by shivanshtalwar** [reply]
Okay so maybe try using streaming id for rtcrenderer map instead of track id in onTrack stream handler

**Reply by ParthSolankii** [reply]
> is it still happening?

I am also having a similar problem, both the camera window are showing on the caller's mobile, But the receiver is not able to see the caller's window.
https://github.com/flutterjanus/flutter_janus_client/issues/151

**Reply by shivanshtalwar0** [reply]
in the latest commits this should be fixed let me know if you still face this

**Reply by shivanshtalwar0** [reply]
so https://github.com/flutterjanus/flutter_janus_client/issues/151  this fixed for you or not?

**Comment by ParthSolankii** [comment]
I am facing another issue in this while calling, that needs to be fixed.

Not able to call again after disconnect the call.

**Reply by shivanshtalwar0** [reply]
not sure of your codebase, but from what i understand if your session/ handle is detroyed using the same session and handle will not work, so either make sure session is not destroyed and attach a new handle every time you want to call again or update the existing handle with new handle value

**Comment by ParthSolankii** [comment]
What will be the best way to achieve continuously calling after disconnect
the call ?

i am sending hang-up data on hangup,

**Reply by shivanshtalwar0** [reply]
yes hangup tears up the internal peer-connection and also destroys the resources of that handle so to let users have call again 
initialize pluginHandle again something like that by
```dart
initializePlugin(){
this.pluginHandle=await this.session.attach<VideoRoomPlugin>()
//rest of hooks associated to this.pluginHanlde like onMessage,onTrack and so on
}
``` 
so everytime you destroy or hangup make sure you call `initialize()` so all the state of plugin is fresh
